ngraph/core/src/op/ctc_greedy_decoder_seq_len.cpp

```diff
@@ -0,0 +1,168 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#include "ngraph/op/ctc_greedy_decoder_seq_len.hpp"
+
+using namespace std;
+using namespace ngraph;
+
+NGRAPH_RTTI_DEFINITION(op::v6::CTCGreedyDecoderSeqLen, "CTCGreedyDecoderSeqLen", 6);
+
+op::v6::CTCGreedyDecoderSeqLen::CTCGreedyDecoderSeqLen(const Output<Node>& input,
+                                                       const Output<Node>& seq_len,
+                                                       const bool merge_repeated,
+                                                       const element::Type& classes_index_type,
+                                                       const element::Type& sequence_length_type)
+    : Op({input, seq_len})
+    , m_merge_repeated(merge_repeated)
+    , m_classes_index_type(classes_index_type)
+    , m_sequence_length_type(sequence_length_type)
+{
+    constructor_validate_and_infer_types();
+}
+
+op::v6::CTCGreedyDecoderSeqLen::CTCGreedyDecoderSeqLen(const Output<Node>& input,
+                                                       const Output<Node>& seq_len,
+                                                       const Output<Node>& blank_index,
+                                                       const bool merge_repeated,
+                                                       const element::Type& classes_index_type,
+                                                       const element::Type& sequence_length_type)
+    : Op({input, seq_len, blank_index})
+    , m_merge_repeated(merge_repeated)
+    , m_classes_index_type(classes_index_type)
+    , m_sequence_length_type(sequence_length_type)
+{
+    constructor_validate_and_infer_types();
+}
+
+void op::v6::CTCGreedyDecoderSeqLen::validate_and_infer_types()
+{
+    const auto& logits_pshape = get_input_partial_shape(0);
```

Comment:
Please add `NGRAPH_OP_SCOPE` macro. Examples you can find here: https://github.com/openvinotoolkit/openvino/pull/3730